system/Session/Session.php

```diff
@@ -156,49 +170,45 @@ class Session implements SessionInterface
     protected $sidRegexp;
 
+    /**
+     * Session Config
+     */
+    protected SessionConfig $config;
+
     /**
      * Constructor.
      *
      * Extract configuration settings and save them here.
      */
-    public function __construct(SessionHandlerInterface $driver, App $config)
+    public function __construct(SessionHandlerInterface $driver, SessionConfig $session)
     {
         $this->driver = $driver;
 
-        /** @var SessionConfig|null $session */
-        $session = config('Session');
+        $this->config = $session;
 
         // Store Session configurations
-        if ($session instanceof SessionConfig) {
-            $this->sessionDriverName        = $session->driver;
-            $this->sessionCookieName        = $session->cookieName ?? $this->sessionCookieName;
-            $this->sessionExpiration        = $session->expiration ?? $this->sessionExpiration;
-            $this->sessionSavePath          = $session->savePath;
-            $this->sessionMatchIP           = $session->matchIP ?? $this->sessionMatchIP;
-            $this->sessionTimeToUpdate      = $session->timeToUpdate ?? $this->sessionTimeToUpdate;
-            $this->sessionRegenerateDestroy = $session->regenerateDestroy ?? $this->sessionRegenerateDestroy;
-        } else {
-            // `Config/Session.php` is absence
-            $this->sessionDriverName        = $config->sessionDriver;
-            $this->sessionCookieName        = $config->sessionCookieName ?? $this->sessionCookieName;
-            $this->sessionExpiration        = $config->sessionExpiration ?? $this->sessionExpiration;
-            $this->sessionSavePath          = $config->sessionSavePath;
-            $this->sessionMatchIP           = $config->sessionMatchIP ?? $this->sessionMatchIP;
-            $this->sessionTimeToUpdate      = $config->sessionTimeToUpdate ?? $this->sessionTimeToUpdate;
-            $this->sessionRegenerateDestroy = $config->sessionRegenerateDestroy ?? $this->sessionRegenerateDestroy;
-        }
+        $this->sessionDriverName        = $session->driver;
+        $this->sessionCookieName        = $session->cookieName;
+        $this->sessionExpiration        = $session->expiration;
+        $this->sessionSavePath          = $session->savePath;
+        $this->sessionMatchIP           = $session->matchIP;
+        $this->sessionTimeToUpdate      = $session->timeToUpdate;
+        $this->sessionRegenerateDestroy = $session->regenerateDestroy;
+
+        /** @var App $config */
+        $config = config('App');
 
         // DEPRECATED COOKIE MANAGEMENT
         $this->cookiePath     = $config->cookiePath ?? $this->cookiePath;
         $this->cookieDomain   = $config->cookieDomain ?? $this->cookieDomain;
         $this->cookieSecure   = $config->cookieSecure ?? $this->cookieSecure;
         $this->cookieSameSite = $config->cookieSameSite ?? $this->cookieSameSite;
 
-        /** @var CookieConfig $cookie */
+        /** @var CookieConfig|null $cookie */
```

Comment:
It seems we can't remove the Session's dependency on the `Config\App` without first removing the cookie items in `Config\App` and making the `Config\Cookie` mandatory...

Reply:
If everyone would be okay with removing session items from `Config\App` in 4.4, then I see no reason why we can't do the same with cookies.

Reply:
I sent a PR #7011 to remove cookie items.
Would everyone be okay with removing items from `Config\App` in 4.4?